core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreWebViewClient.kt

```diff
@@ -102,7 +103,11 @@ open class CoreWebViewClient(
             flags = Intent.FLAG_ACTIVITY_NO_HISTORY
             addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
           }
-          callback.openExternalUrl(intent)
+          val errorMessage = if (extension == "application/pdf")
+            R.string.no_pdf_application_installed
+          else
```

Comment:
The whole function is wrong:
* Either this is an external link http/https and then you pass out to browser whatever
* Or this is an internal link and you pass to kiwix to open and then kiwix decides based on the mime-type stored in the ZIM to open itself or to pass to the OS to open via a third party app